### PR TITLE
[bitnami/redis-cluster] add loadbalancer option to default redis service

### PR DESCRIPTION
**Pull request by doug-ba**
**Description of the change**

Adds the options to have the default "redis" service created as type "LoadBalancer" and to optionally specify a static loadbalancer IP address.
**Benefits**

For my use case with externalAccess enabled this will allow me to create a single DNS record via external-dns which will point to all cluster nodes for hiredis-vip clients to use on their initial connection to get cluster node information.

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by miguelaeh** [comment]
Hi @doug-ba ,
I would like to understand the use case better. The changes you are implementing are for the default service, that is not intended to be exposed, because when your client is redirected, it will be redirected to the IP of a Pod, that will not be reachable from any entity outside the Kubernetes cluster.

**Comment by doug-ba** [comment]
The client is redirected at the Redis-cluster layer, not the kubernetes layer. The Redis-cluster nodes are aware of and redirect to the load balancer services already.

I’ve tested this change and it behaves as I expect it to, redirecting to the load balancer ip addresses.

**Comment by doug-ba** [comment]
```
$ kubectl -n redis get services
NAME                           TYPE           CLUSTER-IP       EXTERNAL-IP    PORT(S)                          AGE
redis-redis-cluster            LoadBalancer   172.24.42.221    10.63.24.118   6379:30468/TCP                   19h
redis-redis-cluster-0-svc      LoadBalancer   172.24.4.136     10.63.26.120   6379:30146/TCP,16379:32101/TCP   19h
redis-redis-cluster-1-svc      LoadBalancer   172.24.0.231     10.63.24.89    6379:30856/TCP,16379:31603/TCP   19h
redis-redis-cluster-2-svc      LoadBalancer   172.24.108.186   10.63.26.221   6379:32760/TCP,16379:32754/TCP   19h
redis-redis-cluster-3-svc      LoadBalancer   172.24.39.112    10.63.27.67    6379:32087/TCP,16379:32756/TCP   19h
redis-redis-cluster-4-svc      LoadBalancer   172.24.114.147   10.63.24.106   6379:31134/TCP,16379:32710/TCP   19h
redis-redis-cluster-5-svc      LoadBalancer   172.24.52.157    10.63.24.68    6379:31500/TCP,16379:30668/TCP   19h
redis-redis-cluster-headless   ClusterIP      None             <none>         6379/TCP,16379/TCP               19h
```
```$ redis-cli -c -h 10.63.24.118
10.63.24.118:6379> get foo
-> Redirected to slot [12182] located at 10.63.27.67:6379
"bar"
10.63.27.67:6379> get bar
-> Redirected to slot [5061] located at 10.63.24.68:6379
"foo"
10.63.24.68:6379>
```

**Comment by doug-ba** [comment]
@miguelaeh I forgot to @ you in my comments above.

**Comment by miguelaeh** [comment]
Hi @doug-ba ,
I missed that you are using external access. When using external access, the Redis nodes are configured to use the LocalBalancerIPs of the services (that is the reason why you need to perform an upgrade setting the IPs), so my previous comment that not apply to your use case.
So, I think I got your point now, you want to have a "single entry point" to the Redis cluster, as you are using external access your clients will be redirected to the LoadBalancerIPs, so you can make the requests to a single point and then be redirected to the correct Redis node.

File: bitnami/redis-cluster/values-production.yaml [hunk 1]
```diff
@@ -265,7 +265,13 @@ service:
...
   annotations: {}
   labels: {}
-
+  ## Set the service type.
+  ## Setting this to LoadBalancer may require corresponding service annotations for loadbalancer creation to succeed.
+  ## Currently supported types are ClusterIP (default) and LoadBalancer
+  type: ClusterIP
+  ## If service.type is LoadBalancer, request a specific static IP address if supported by the cloud provider.
+  ## otherwise leave blank
+  # loadBalancerIP:
```
Comment (miguelaeh) [inline bitnami/redis-cluster/values-production.yaml:274]: Please add a space below to separate the sections.

File: bitnami/redis-cluster/values.yaml [hunk 2]
```diff
@@ -265,7 +265,13 @@ service:
...
   annotations: {}
   labels: {}
-
+  ## Set the service type.
+  ## Setting this to LoadBalancer may require corresponding service annotations for loadbalancer creation to succeed.
+  ## Currently supported types are ClusterIP (default) and LoadBalancer
+  type: ClusterIP
+  ## If service.type is LoadBalancer, request a specific static IP address if supported by the cloud provider.
+  ## otherwise leave blank
+  # loadBalancerIP:
```
Comment (miguelaeh) [inline bitnami/redis-cluster/values.yaml:274]: The same here

File: bitnami/redis-cluster/values-production.yaml (hunk 1 above)
Comment (doug-ba) [reply bitnami/redis-cluster/values-production.yaml:276]: @miguelaeh added line breaks.

File: bitnami/redis-cluster/values.yaml (hunk 2 above)
Comment (doug-ba) [reply bitnami/redis-cluster/values.yaml:276]: @miguelaeh added line breaks.

**Comment by doug-ba** [comment]
@miguelaeh I added line breaks to the areas you commented on, hopefully I understood what you were looking for.

**Comment by doug-ba** [comment]
@miguelaeh , Does your CI/CD run on its own or does it need to be triggered?

**Comment by miguelaeh** [comment]
Hi @doug-ba ,
Sorry for the time this is taking, I run the jobs yesterday but our pipeline was unestable, I will run it now again.

**Comment by doug-ba** [comment]
@miguelaeh Thanks a bunch!

**Comment by doug-ba** [comment]
@miguelaeh how do we get the release built?

**Comment by miguelaeh** [comment]
Hi @doug-ba ,
Just execute:
```
helm repo update
```

And the new version should have your changes

**Pull request merged**